core/Pos/Crypto/SafeSigning.hs

```diff
@@ -65,12 +66,22 @@ encToPublic (EncryptedSecretKey sk) = PublicKey (CC.toXPub sk)
 noPassEncrypt :: SecretKey -> EncryptedSecretKey
 noPassEncrypt (SecretKey k) = EncryptedSecretKey k
 
-signRaw' :: PassPhrase -> EncryptedSecretKey -> ByteString -> Signature Raw
-signRaw' (PassPhrase pp) (EncryptedSecretKey sk) x =
-    Signature (CC.sign pp sk x)
-
-sign' :: Bi a => PassPhrase -> EncryptedSecretKey -> a -> Signature a
-sign' pp sk = coerce . signRaw' pp sk . BSL.toStrict . Bi.encode
+signRaw' :: Maybe SignTag
+         -> PassPhrase
+         -> EncryptedSecretKey
+         -> ByteString
+         -> Signature Raw
+signRaw' mbTag (PassPhrase pp) (EncryptedSecretKey sk) x =
+    Signature (CC.sign pp sk (tag <> x))
+  where
+    tag = case mbTag of
```

Comment:
I changed it to `maybe` in the next commit (which will be a part of CSL-990). I don't like using folds on `Maybe` so I'd rather not use `foldMap`.